alibi/saving.py

```diff
@@ -232,14 +232,25 @@ def _save_AnchorText(explainer: 'AnchorText', path: Union[str, os.PathLike]) ->
     explainer.perturbation = perturbation
 
 
-def _save_KernelShap(explainer: 'KernelShap', path: Union[str, os.PathLike]) -> None:
-    # TODO: save internal shap objects using native pickle?
+def _save_Shap(explainer: Union['KernelShap', 'TreeShap'], path: Union[str, os.PathLike]) -> None:
+    # set the internal explainer object to avoid saving it. The internal explainer
+    # object is recreated when in the `reset_predictor` function call.
+    _explainer = explainer._explainer
+    explainer._explainer = None
```

Comment:
Just noting that this will need to be considered too when fixing https://github.com/SeldonIO/alibi/issues/830